subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -145,29 +148,29 @@ public InputStream apply(@Nullable String input) {
         }
 
         @Override
-        public void execute(PrintStream out) {
+        public void execute(Logger logger) {
             if (isWelcomeMessageEnabled()) {
                 File markerFile = getMarkerFile();
 
-                if (!markerFile.exists()) {
-                    out.println();
-                    out.print("Welcome to Gradle " + gradleVersion.getVersion() + "!");
+                if (!markerFile.exists() && logger.isLifecycleEnabled()) {
+                    logger.lifecycle("");
+                    logger.lifecycle("Welcome to Gradle " + gradleVersion.getVersion() + "!");
 
                     String featureList = readReleaseFeatures();
 
                     if (StringUtils.isNotBlank(featureList)) {
-                        out.println();
-                        out.println();
-                        out.println("Here are the highlights of this release:");
-                        out.print(featureList);
+                        logger.lifecycle("");
+                        logger.lifecycle("");
```

Comment:
Since the `print()` in line 154 is now effectively a `println()` one blank line should be removed here.

```diff

  Welcome to Gradle 42.0!

+
  Here are the highlights of this release:
   - foo
   - bar
```

Reply:
Restored the behavior and updated the corresponding test.